### PR TITLE
god view app

### PR DESCRIPTION
**Pull request by ZappoMan**
This script provides a god view, that moves your avatar above the world looking down, you can use keyboard or mouse to then move above the world. This is part of a prototype for a mobile 2D movement concept, but is also useful in desktop mode.

### CODE REVIEW

**Comment by ZappoMan** [comment]
Test plan:
- run this build
- in Running scripts, search for godView - run it.
- you should see a tablet and/or toolbar icon for "God View"
- when you enable god view, you can use keyboard arrows to move around, or you can click on the screen with your mouse to move above the location of the world, ending god view drops you back down to the level you were when you entered god view.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/368/HighFidelity-Beta-PR10607-60048a231f1409e21857c66a0651bdb397613a47.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/368/HighFidelity-Beta-PR10607-60048a231f1409e21857c66a0651bdb397613a47.exe

**Comment by CainFoool** [comment]
Will test.

**Comment by CainFoool** [comment]
# Test Passed 

God View works as expected; however IMO it should teleport you back to the original location you started at when you exit - not just the level but X and Z too; unless this is meant to be some form of quick teleportation.

Additionally, I would suggest using WASD as an additional way to move around as switching around with arrows keys, while it's logical, for gaming purposes WASD makes more sense.

File: scripts/system/controllers/godView.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    }
+}
+
+function mousePress(event) {
+    if (godView) {
+        var pickRay = Camera.computePickRay(event.x, event.y);
+        Vec3.print("pr.o:", pickRay.origin);
+        Vec3.print("pr.d:", pickRay.direction);
+        Vec3.print("c.p:", Camera.position);
```
Comment (Atlante45) [inline scripts/system/controllers/godView.js]: Do you want to keep that debug?

File: scripts/system/controllers/godView.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        var pointingAt = Vec3.sum(pickRay.origin, Vec3.multiply(pickRay.direction,300));
+        var moveToPosition = { x: pointingAt.x, y: MyAvatar.position.y, z: pointingAt.z };
+        moveTo(moveToPosition);
+    }
+}
+
+
+function startGodView() {
+    if (!godView) {
+        Camera.mode = "first person";
```
Comment (Atlante45) [inline scripts/system/controllers/godView.js]: I don't understand why you switch to first person here.
Seems like you should save the current mode here to restore it in `endGodView` instead.

Comment (ZappoMan) [inline scripts/system/controllers/godView.js]: good point. yes I should do that

File: scripts/system/controllers/godView.js (hunk 1 above)
Comment (ZappoMan) [inline scripts/system/controllers/godView.js]: will fix

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1381/HighFidelity-Beta-PR10607-10341af819eddcbecfc44102b92f6a1ba819abb9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1381/HighFidelity-Beta-PR10607-10341af819eddcbecfc44102b92f6a1ba819abb9.dmg

**Comment by YuppyPlays** [comment]
Will test once built :)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1388/HighFidelity-Beta-PR10607-c3c9df0cd0dd521e37f99679b029647c437b53e5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1388/HighFidelity-Beta-PR10607-c3c9df0cd0dd521e37f99679b029647c437b53e5.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

run this build **Check**
in Running scripts, search for godView - run it. **Check**
you should see a tablet and/or toolbar icon for "God View" **Check**
when you enable god view, you can use keyboard arrows to move around, or you can click on the screen with your mouse to move above the location of the world, ending god view drops you back down to the level you were when you entered god view. **Pass, arrows do a snap like movment by a small amount up,down,left,right as expected & the mouse changes the location relative to where it's clicked**
verify that you return to the camera mode you were in when you started, namely if you were in first person, you return to first person when leaving god mode, if you were in third person, you should return to that. **Pass, tested in third person and first, returns to the same state for each as well as amount scroll out in third person**

**Pass!** :D Please let me know if there is anything more I can do/provide to assist.

**Pull request merged**